x/evm/keeper/invariants.go

```diff
@@ -30,18 +30,19 @@ func (k Keeper) BalanceInvariant() sdk.Invariant {
 			count int
 		)
 
-		k.accountKeeper.IterateAccounts(ctx, func(account authexported.Account) bool {
+		k.accountKeeper.IterateAccounts(ctx, func(account authtypes.AccountI) bool {
 			ethAccount, ok := account.(*ethermint.EthAccount)
 			if !ok {
 				// ignore non EthAccounts
 				return false
 			}
 
 			evmDenom := k.GetParams(ctx).EvmDenom
-			accountBalance := ethAccount.GetCoins().AmountOf(evmDenom)
+
+			accountBalance := k.bankKeeper.GetBalance(ctx, ethAccount.GetAddress(), evmDenom)
```

Comment:
question: where in the code does it update both evm & bankkeeper balance when performing a balance transfer?

Reply:
when the dirty state object is finalized and commited